### PR TITLE
Pagination bug fixes

### PR DESCRIPTION
**Pull request by LTA-Thinking**
## Description
Fixes two bugs with pagination:
1. Fixes an issue where if a search is done with _include and descending _sort with a _count value less than the total number of matched resources but greater than the number of matched resources that contain the sort field and their included resources than the result will not contain a next link and the total count will be wrong.
2. Fixes an issue where the summary and total count of resources returned when using reverse chain will include deleted resources.

Example data that demonstrates these bugs is also included, although the other changes need to be reverted to show the issues.

## Related issues
Addresses [Bug 124924](https://microsofthealth.visualstudio.com/Health/_workitems/edit/124924): Pagination bugs

## Testing
Additional E2E tests have been added to cover these scenarios.

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

**Review by fhibf — APPROVED** [review]

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/IncludeSearchTests.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@
 using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 using System.Web;
+using DotLiquid.Tags;
```
Comment (brendankowitz) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/IncludeSearchTests.cs:13]: 🤔

**Review by brendankowitz — DISMISSED** [review]

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/IncludeSearchTests.cs (hunk 1 above)
Comment (LTA-Thinking) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/IncludeSearchTests.cs:13]: Ug, that is probably from an autocomplete when I was adding the word tag to the search query.

**Review by fhibf — APPROVED** [review]

**Pull request merged**